### PR TITLE
Disable Non ACA Individual market Features

### PR DESCRIPTION
**Pull request by jacobkagon**
### Master Pivotal ticket
* https://www.pivotaltracker.com/n/projects/2481179/stories/178611906

### Child Redmine ticket(s)
* ()
* ()

### Local build result

```
bundle exec rake parallel:spec[4]
bundle exec cucumber
```

### Latest rebase/merge tag
*

---

### Data ticket(s) Followup
* (redmine links here - optional)

### Related Pull Requests
* (github links here - optional)

---

#### Peer Review
* (For code review)

#### Functional Testing
* (For testing locally)

#### Deployment
* (for release manager and/or build manager)

### CODE REVIEW

**Review by bill-transue — APPROVED** [review]
Approved, thanks for changes @jacobkagon!

File: system/config/templates/features/enroll_app/hbx_admin_config.yml [hunk 1]
```diff
@@ -13,3 +13,5 @@ registry:
           is_enabled: true
         - key: :inbox_tab
```
Comment (ZASMan) [inline system/config/templates/features/enroll_app/hbx_admin_config.yml:14]: Why did you add a new feature? Shouldn't you just be using `EnrollRegistry.feature_enabled?(:aca_individual_market)`?

File: config/client_config/dc/system/config/templates/features/enroll_app/hbx_admin_config.yml [hunk 2]
```diff
@@ -13,3 +13,5 @@ registry:
           is_enabled: true
         - key: :inbox_tab
           is_enabled: true
+        - key: :aca_individual_market_feature
```
Comment (bill-transue) [inline config/client_config/dc/system/config/templates/features/enroll_app/hbx_admin_config.yml:16]: I missed this new item on my first pass through, we don't need this new config item. We're already setting IVL market on and off elsewhere. Use the existing check instead: `EnrollRegistry.feature_enabled?(:aca_individual_market)`.

File: app/views/ui-components/v1/navs/_primary_nav.html.slim [hunk 3]
```diff
@@ -19,8 +19,9 @@
...
               li
                 = render partial: 'exchanges/hbx_profiles/shared/nav_link', locals: {tab_name: 'identity_verification', link_path: main_app.identity_verification_exchanges_hbx_profiles_path, display_name: 'Identity Verification', active_tab: "identity_verification", remote: false, pundit_object: HbxProfile, pundit_method: :access_identity_verification_sub_tab?}
-              li
-                = render partial: 'exchanges/hbx_profiles/shared/nav_link', locals: {tab_name: 'dc-resident-application', link_path: main_app.begin_resident_enrollment_exchanges_residents_path(original_application_type: 'paper'), display_name: "#{EnrollRegistry[:enroll_app].settings(:site_key).item.upcase} Resident Application", active_tab: "dc-resident-application", remote: false, pundit_object: ResidentRole, pundit_method: :begin_resident_enrollment?}
+              - if EnrollRegistry.feature_enabled?(:aca_individual_market_feature)
```
Comment (bill-transue) [inline app/views/ui-components/v1/navs/_primary_nav.html.slim:22]: Instead of adding in a new check let's use the existing one: `EnrollRegistry.feature_enabled?(:aca_individual_market)`.

File: app/models/effective/datatables/family_data_table.rb [hunk 4]
```diff
@@ -57,17 +57,19 @@ class FamilyDataTable < Effective::MongoidDatatable
...
-                ['Edit APTC / CSR', edit_aptc_csr_path(family_id: row.id, person_id: row.primary_applicant.person.id),
-                 aptc_csr_link_type(row, pundit_allow(Family, :can_update_ssn?))],
-                ['Paper', resume_enrollment_exchanges_agents_path(person_id: row.primary_applicant.person.id, original_application_type: 'paper'), 'static'],
-                ['Phone', resume_enrollment_exchanges_agents_path(person_id: row.primary_applicant.person.id, original_application_type: 'phone'), 'static'],
-                ['Transition Family Members', transition_family_members_insured_families_path(family: row.id, family_actions_id: "family_actions_#{row.id}"),
-                 transition_family_members_link_type(row, pundit_allow(Family, :can_transition_family_members?)) ? 'ajax' : 'disabled']
-
-            ]
-          end
+          dropdown += if individual_market_is_enabled?
```
Comment (ZASMan) [inline app/models/effective/datatables/family_data_table.rb:60]: I'm a little rusty but what exactly are you trying to do here, remove the `transition_family_members_link_type` unless aca individual market is enabled?

It would be better to do something like

```
dropdown << ['Transition Family Members', transition_family_members_insured_families_path(family: row.id, family_actions_id: "family_actions_#{row.id}"),
                           transition_family_members_link_type(row, pundit_allow(Family, :can_transition_family_members?)) ? 'ajax' : 'disabled']
                        ] if aca_individual_market_feature_is_enabled?
 ```
Just in case more options are added

**Review by bill-transue — CHANGES_REQUESTED** [review]
We shouldn't add redundant configuration items to the resource registry files.

**Review by ZASMan — CHANGES_REQUESTED** [review]

File: app/views/ui-components/v1/navs/_primary_nav.html.slim [hunk 5]
```diff
@@ -19,7 +19,7 @@
                 = render partial: 'exchanges/hbx_profiles/shared/nav_link', locals: {tab_name: 'new-consumer-application', link_path: main_app.begin_consumer_enrollment_exchanges_agents_path(original_application_type: 'phone'), display_name: 'New Consumer Application', active_tab: "new-consumer-application", remote: false, pundit_object: ConsumerRole, pundit_method: :access_new_consumer_application_sub_tab?}
               li
                 = render partial: 'exchanges/hbx_profiles/shared/nav_link', locals: {tab_name: 'identity_verification', link_path: main_app.identity_verification_exchanges_hbx_profiles_path, display_name: 'Identity Verification', active_tab: "identity_verification", remote: false, pundit_object: HbxProfile, pundit_method: :access_identity_verification_sub_tab?}
-              - if EnrollRegistry.feature_enabled?(:aca_individual_market_feature)
+              - if EnrollRegistry.feature_enabled?(:non_aca_individual_market_feature)
```
Comment (bill-transue) [inline app/views/ui-components/v1/navs/_primary_nav.html.slim]: This condition isn't going to work here. The partial needs to be rendered regardless of the feature.

**Review by bill-transue — CHANGES_REQUESTED** [review]
This has gotten worse since I last looked at it. Dropdown has an array of arrays so you need to work with the array within the array. Also you need to render the dropdown regardless whether feature is enabled or not.

**Review by bill-transue — APPROVED** [review]
Thanks for looking into my concerns @jacobkagon, great work!

**Comment by harshared** [comment]
@jacobkagon please resolve conflicts
